File: Algorithm/DataStructure/stackAndqueue/array_stack.py
class ArrayStack:
    def __init__(self) -> None:
        self._stack: list[int] = []

    # 获取栈的大小
    def size(self) -> int:
        return len(self._stack)

    # 判断栈是否为空
    def is_empty(self) -> bool:
        return len(self._stack) == 0

    # 入栈
    def push(self, item: int):
        self._stack.append(item)

    # 出栈
    def pop(self) -> int:
        if self.is_empty():
            raise IndexError("栈为空")
        return self._stack.pop()

    # 访问栈顶元素
    def peek(self) -> int:
        if self.is_empty():
            raise IndexError("栈为空")
        return self._stack[-1]

    # 转换为列表
    def to_list(self) -> list[int]:
        return self._stack

File: Algorithm/DataStructure/stackAndqueue/ArrayStack.java
package Algorithm.DataStructure.stackAndqueue;

import java.util.ArrayList;

public class ArrayStack {
    private ArrayList<Integer> stack;

    public ArrayStack(){
        stack = new ArrayList<>();
    }

    // 栈的长度
    public int size(){
        return stack.size();
    }

    // 判断栈是否为空
    public boolean isEmpty(){
        return stack.size() == 0;
    }

    // 入栈
    public void push(int num){
        this.stack.add(num);
    }

    // 出栈
    public int pop(){
        if (this.isEmpty()){
            throw new IndexOutOfBoundsException();
        }
        return this.stack.remove(this.size() - 1);
    }

    // 访问栈顶元素
    public int peek(){
        if (this.isEmpty()){
            throw new ArrayIndexOutOfBoundsException();
        }
        return this.stack.get(this.size() - 1);
    }

    // 将List转化为Array返回
    public Object[] toArray(){
        return stack.toArray();
    }
}
